src/averageAges.js

```diff
@@ -38,6 +50,20 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const women = people.filter(person =>
+    person.sex === 'f' && (!withChildren || hasChildren(person, people))
+  );
+
+  const womenAgesSum = women.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+
+  const womenCount = women.length;
+
+  return womenCount > 0 ? womenAgesSum / womenCount : 0;
```

Comment:
we can move it to some helper function and reuse it here and in the prev func too